drivers/can/can_mcan.c

```diff
@@ -562,10 +610,19 @@ void can_mcan_line_0_isr(const struct device *dev)
 			LOG_ERR("Access to reserved address");
 		}
 
-		if (ir & CAN_MCAN_IR_MRAF) {
+		if ((ir & CAN_MCAN_IR_MRAF) != 0U) {
 			LOG_ERR("Message RAM access failure");
 		}
 
+		if ((ir & (CAN_MCAN_IR_PEA | CAN_MCAN_IR_PED)) != 0U) {
+			LOG_WRN("Protocol Error");
```

Comment:
I don't think a log message is warranted here. That could lead to quite a lot of log messages. This is why the CAN stats was introduced in the first place.